engine/src/main/java/com/datatorrent/stram/StreamingAppMasterService.java

```diff
@@ -750,6 +752,11 @@ private void execute() throws YarnException, IOException
         expiryTime = StramUserLogin.refreshTokens(tokenLifeTime, FileUtils.getTempDirectoryPath(), applicationId, conf, hdfsKeyTabFile, credentials, rmAddress, true);
       }
 
+      if (System.currentTimeMillis() > nodeReportUpdateTime) {
```

Comment:
It has to be in the loop, but no need to call it multiple times. The millis are used in several places.